### ISSUE TITLE
Merge requirements dev.txt and test.txt, include prod_without_hashes.txt

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#2596]
Our requirements are a mess. #2300 broke new installs (using docker or manual) because `prod_without_hashes.txt` is not included anywhere. It's used by tox and our deploy scripts so travis and prod/dev/stage continue to work normally... We should include it in `dev.txt`.

In addition, I'm not sure it's worth separating `dev.txt` and `test.txt` anymore, it adds complexity and I don't see a good reason to do it.

**Comment by EnTeQuAk** [comment]
Also, the ways we do `pip install` in our `Makefile` are different from what we're using in docker. I think in a dream world all tools would just use our `Makefile` and call `update_deps` so that we have one single source of "that's how it's installed, that's where index servers are defined" etc.

**Comment by diox** [comment]
> Also, the ways we do pip install in our Makefile are different from what we're using in docker. I think in a dream world all tools would just use our Makefile and call update_deps so that we have one single source of "that's how it's installed, that's where index servers are defined" etc.

This, a thousand times.

**Comment by muffinresearch** [comment]
In case it helps anyone else find this the error message seen here when hitting this issue is:

```
ImportError: No module named session_csrf
```

**Comment by muffinresearch** [comment]
As a hacky workaround you can run this to update the deps on docker:

```
docker exec -t -i "addonsserver_worker_1" pip install --no-deps --exists-action=w -r requirements/prod_without_hash.txt --find-links https://pyrepo.stage.mozaws.net/olympia/ --find-links https://pyrepo.stage.mozaws.net/ --no-index
docker exec -t -i "addonsserver_web_1" pip install --no-deps --exists-action=w -r requirements/prod_without_hash.txt --find-links https://pyrepo.stage.mozaws.net/olympia/ --find-links https://pyrepo.stage.mozaws.net/ --no-index
```

The work + web services then need to be restarted either via supervisorctl on each running container or by just running `docker-compose stop && docker-compose up -d` the latter is probably more straight-forward.

Running `docker-compose pull` again will cause the bug to re-occur until this issue is fixed.

**Comment by muffinresearch** [comment]
I think it would be worth doing a quick fix to sort out restoring the dev environment and then a more ideal solution to deps handling should happen after that.

**Comment by muffinresearch** [comment]
Marking this a priority one since our development environment is broken and that's impacting devs and also contributors.

**Comment by davehunt** [comment]
I'm probably doing something wrong, but the workaround isn't working for me. I added https://github.com/davehunt/addons-server/commit/dd9b7eebddf355c1da29587f334ed62a72788f1f to my branch with experimental CircleCI builds ([docker exec is not supported](https://circleci.com/docs/docker/#docker-exec)). Any other suggestions that might unblock me?

**Comment by muffinresearch** [comment]
@davehunt - not sure. We really need a better fix for this. You could try running those commands after  `scripts/setup-docker.sh`

**Comment by andymckay** [comment]
My local docker instance is hosed now too :(

**Comment by EnTeQuAk** [comment]
This should be fixed now that https://github.com/mozilla/addons-server/pull/2635 landed.

Please feel free to reopen if something is still failing.

**Comment by andymckay** [comment]
This still occurs for me.

The reason is that after doing a build, the `prod_without_hashes` dependencies get written to `/code/src`. Later on the docker-compose file overwrites `/code`. @EnTeQuAk and others, I suspect you have those libraries in your addons-server checkout locale.

```
docker build -t amo-test .
[...]
docker run -it amo-test:latest
[root@7eb53e4328a5 code]# cat /usr/lib/python2.7/site-packages/django-session-csrf.egg-link
/code/src/django-session-csrf
.[root@7eb53e4328a5 code]# ls /code/src/django-session-csrf/
django_session_csrf.egg-info  MANIFEST.in  requirements.txt  session_csrf
LICENSE                       README.rst   runtests.sh       setup.py
```
All good until you run it from docker-compose: https://github.com/mozilla/addons-server/blob/master/docker-compose.yml#L5-L6

```
addons-server docker-compose up -d                                      master  ✭ ✱ 
addonsserver_redis_1 is up-to-date
addonsserver_rabbitmq_1 is up-to-date
addonsserver_elasticsearch_1 is up-to-date
addonsserver_mysqld_1 is up-to-date
addonsserver_memcached_1 is up-to-date
addonsserver_worker_1 is up-to-date
addonsserver_web_1 is up-to-date
addonsserver_nginx_1 is up-to-date
addons-server make shell                                                master  ✭ ✱ 
docker exec -t -i "addonsserver_web_1" bash
[root@87a879eb37ef code]# ls /code/src
olympia  olympia.egg-info
```

I think we should encourage the `prod_without_hashes` to install somewhere else.

**Comment by andymckay** [comment]
I'm getting:

```
amo master addons-server $ tail logs/docker.log 
    __import__(name)
  File "/code/src/olympia/api/models.py", line 6, in <module>
    from aesfield.field import AESField
  File "/usr/lib/python2.7/site-packages/aesfield/field.py", line 6, in <module>
    from m2secret import Secret
  File "/usr/lib/python2.7/site-packages/m2secret.py", line 36, in <module>
    from M2Crypto import EVP
  File "/usr/lib64/python2.7/site-packages/M2Crypto/__init__.py", line 22, in <module>
    import __m2crypto
ImportError: libssl.so.1.0.0: cannot open shared object file: No such file or directory
```

The earlier error was masking this one.

**Comment by tsl143** [comment]
i am getting same error as @andymckay

**Comment by EnTeQuAk** [comment]
Should be fixed with https://github.com/mozilla/addons-server/pull/2684

**Comment by tsl143** [comment]
Now I get following error.
![screenshot from 2016-05-18 14 12 43](https://cloud.githubusercontent.com/assets/5318732/15352629/b71c0322-1d02-11e6-827e-2bd5c041a8e4.png)

**Comment by diox** [comment]
@tsl143 : can you file a new issue for this? thanks

**Comment by tsl143** [comment]
@diox Done, thanks
